### DISCUSSION TITLE
CPU impact calculations

### DISCUSSION THREAD

**Discussion by consithe1** [Boavizta/boaviztapi#198] [Q&A] [answered]
Hello!

Going through the documentation of BoaviztAPI, I was trying to reproduce the impact calculations for the CPU. However, it seems like I am stuck in different ways and I was wondering if you could help me with it please.

1. In the documentation, the die_size_per_core corresponds to the [number of cores divided by the die size](https://github.com/Boavizta/boaviztapi/blob/main/docs/docs/Explanations/components/cpu.md#die_size_per_core-and-core_units).
However, in the code, it seems like it's the other way around: [die_size divided by the number of cores](https://github.com/Boavizta/boaviztapi/blob/9dd02ad0582e47669186184864a3902f2df65c3b/boaviztapi/dto/component/cpu.py#L70) which is logic when you think about the variable name. 
My problem is unfortunately the following one: doesn't matter which formula you take, the die_size_per_core value still never fits with the [results you find](https://github.com/Boavizta/boaviztapi/blob/main/boaviztapi/data/components/cpu_manufacture.csv)
For example, if we take a die_size of 694 mm², a number of cores of 28. By applying the formula, we get a die_size_per_core of around 25mm². Why does your data indicates a die_size_per_core of 0.248 in this case please?

Best regards

**Comment by da-ekchajzer** [comment, marked as answer]
Hello,

Thank you for opening this discussion.

> In the documentation, the die_size_per_core corresponds to the [number of cores divided by the die size](https://github.com/Boavizta/boaviztapi/blob/main/docs/docs/Explanations/components/cpu.md#die_size_per_core-and-core_units).
> However, in the code, it seems like it's the other way around: [die_size divided by the number of cores](https://github.com/Boavizta/boaviztapi/blob/9dd02ad0582e47669186184864a3902f2df65c3b/boaviztapi/dto/component/cpu.py#L70) which is logic when you think about the variable name.

There is a typo in the documentation that we just corrected. Thank you for your report.

Commit : https://github.com/Boavizta/boaviztapi/commit/e659191810daca9db7efda2949bf8be0e93f177e

> My problem is unfortunately the following one: doesn't matter which formula you take, the die_size_per_core value still never fits with the [results you find](https://github.com/Boavizta/boaviztapi/blob/main/boaviztapi/data/components/cpu_manufacture.csv)
> For example, if we take a die_size of 694 mm², a number of cores of 28. By applying the formula, we get a die_size_per_core of around 25mm². Why does your data indicates a die_size_per_core of 0.248 in this case please?

I just found that the cpu internal data in the API is not using mm2, but cm2. This is why you have the issue.

In https://github.com/Boavizta/boaviztapi/blob/main/boaviztapi/data/components/cpu_manufacture.csv, die_size column is in mm2 and die_size_per_core is in cm2


In the case you described : ```694/28 = 24.8 mm2 ≈ 0.248 cm2 ```. 

Since few users provide directly the die_size but use the pre-recorded die_size (which are in cm2) we just found out. Thank you for raising the issue. 

I just updated the documentation and unit in the API response. **You should now provide your die_size in cm2 (/100) and not in mm2.**

The next version of the API (v1 which is still in development) uses only mm2 so this error will no longer occur.

Commit : https://github.com/Boavizta/boaviztapi/commit/f1b8096463b5fd78d9fa308eead14ebc7cca42b1

Could you test and tell us if using cm2 fixes your issue ?

You are also welcome to join our public chat to tell us about your use case : https://chat.boavizta.org/signup_user_complete/?id=97a1cpe35by49jdc66ej7ktrjc

**Reply by consithe1** [reply]
Thank you so much for your reply and answers! I feel a bit stupid to not have thought about the fact that it was in cm² and not in mm² indeed!

I will test as soon as possible to tell you if it fixes the issue.
I might have another question for the CPU energy consumption impact, shoud I create another thread or should I write it here as well?

Good luck with this project and best regards

**Reply by da-ekchajzer** [reply]
Don't feel stupid for a poor documentation on our side ;).

For your other question, you could : 

* Open an issue : https://github.com/Boavizta/boaviztapi/issues
* Open a discussion
* Exchange with us in our public chat : https://chat.boavizta.org/signup_user_complete/?id=97a1cpe35by49jdc66ej7ktrjc
